src/data/buffer/parser/basic.lean

```diff
@@ -2139,4 +2140,221 @@ end
 
 end many
 
+section nat
+/--
+The `val : ℕ` produced by a successful parse of a `cb : char_buffer` is the numerical value
+represented by the string of decimal digits (possibly padded with 0s on the left)
+starting from the parsing position `n` and ending at position `n'`. The number
+of characters parsed in is necessarily `n' - n`.
+-/
+lemma nat_of_done {val : ℕ} (h : nat cb n = done n' val) :
```

Comment:
This is quite a long proof that's only semi-structured. There are some `have`s that make it slightly easier to follow, but could you sketch what you're doing with some comments?

Reply:
Thanks for the suggestions. Can you remark on the comments that I made for `nat_eq_done`? If they're sufficient, I'll do the same for the other lemmas in this PR too.